### PR TITLE
garage: 0.8.0 -> 0.8.1

### PR DESCRIPTION
**Pull request by teutat3s**
###### Description of changes

- Changelog: https://git.deuxfleurs.fr/Deuxfleurs/garage/compare/v0.8.0...v0.8.1
- Fix tests by adding cacert to `checkInputs`
- Use garage version `0_8` by default for nixos tests
- Use same rust feature set as upstream for release binary and tests: https://git.deuxfleurs.fr/Deuxfleurs/garage/src/tag/v0.8.1/nix/compile.nix#L181-L192 https://git.deuxfleurs.fr/Deuxfleurs/garage/src/tag/v0.8.1/default.nix#L40-L52

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by RaitoBezarius — APPROVED** [review]
Sounds good to me.

File: pkgs/tools/filesystems/garage/default.nix [hunk 1]
```diff
@@ -36,20 +40,19 @@ let
...
+    # but in buildFeatures only for version 0.8+, where it's enabled by default.
+    # See: https://garagehq.deuxfleurs.fr/documentation/reference-manual/k2v/
     checkFeatures = [
       "k2v"
       "kubernetes-discovery"
     ] ++
     (lib.optional (lib.versionAtLeast version "0.8") [
       "bundled-libs"
       "sled"
-      "metrics"
```
[the remark this answers is not in the document]
Comment (RaitoBezarius) [reply pkgs/tools/filesystems/garage/default.nix:51]: nvm

**Review by SuperSandro2000 — COMMENTED** [review]
Do we really need the garage_0_7_3 garage_0_8_1 aliases?

Also we should work with nixos release numbers and gradually remove older verisons with new releases.

**Comment by RaitoBezarius** [comment]
> Do we really need the garage_0_7_3 garage_0_8_1 aliases?
> 
> Also we should work with nixos release numbers and gradually remove older verisons with new releases.

Yes because of state versions.

**Comment by RaitoBezarius** [comment]
> Do we really need the garage_0_7_3 garage_0_8_1 aliases?
> 
> Also we should work with nixos release numbers and gradually remove older verisons with new releases.

After the next release, we will remove 0.7.3 I believe.

**Comment by SuperSandro2000** [comment]
Is it expected that aarchz64-darwin fails?

**Comment by teutat3s** [comment]
[This comment](https://github.com/NixOS/nixpkgs/issues/208366#issuecomment-1368255378) from @wegank suggests otherwise

**Comment by teutat3s** [comment]
This is the code of the [failing test](https://logs.nix.ci/?key=nixos/nixpkgs.218843&attempt_id=475d25b5-78bc-4ef2-a569-0b621cefb265): 

https://git.deuxfleurs.fr/Deuxfleurs/garage/src/tag/v0.8.1/src/garage/tests/bucket.rs#L11-L23

**Comment by RaitoBezarius** [comment]
Upstream is investigating and found out a potential race conditions in their tests.

**Comment by teutat3s** [comment]
It seems a possible workaround is: `cargo test -- --test-threads=1` - how would I do that for nixpkgs?

**Comment by teutat3s** [comment]
Tried to address this in https://github.com/NixOS/nixpkgs/pull/218843/files#diff-5927e588b0babe3044e0f517bf32490ed966d770e77432e994756274edc6c788R60-R61 @SuperSandro2000 ready for another review

**Comment by teutat3s** [comment]
Just tested this on `darwin-x86_64` and tests pass there as well. No longer marking as broken.

File: pkgs/tools/filesystems/garage/default.nix [hunk 2]
```diff
@@ -36,24 +40,26 @@ let
...
-      "telemetry-otlp"
       "lmdb"
       "sqlite"
     ]);
 
+    # Workaround until upstream fixes integration test race condition
```
Comment (RaitoBezarius) [inline pkgs/tools/filesystems/garage/default.nix:60]: Do they have an issue upstream on this? If so, we should link it there.

Comment (teutat3s) [reply pkgs/tools/filesystems/garage/default.nix:60]: Just checked again, there's no issue yet. I'll create one and link to it

Comment (teutat3s) [reply pkgs/tools/filesystems/garage/default.nix:60]: Done

**Review by RaitoBezarius — APPROVED** [review]
LGTM

**Pull request merged**